Marlin/src/gcode/motion/G2_G3.cpp

```diff
@@ -320,12 +320,40 @@ void GcodeSuite::G2_G3(const bool clockwise) {
 
       #if ENABLED(ARC_P_CIRCLES)
         // P indicates number of circles to do
-        int8_t circles_to_do = parser.byteval('P');
-        if (!WITHIN(circles_to_do, 0, 100))
+        int8_t circles_to_do = parser.byteval('P', 1);
```

Comment:
This will cause `circles_to_do` to be `1` when no `P` parameter is given. That would be bad. Can you run some tests that don't include a `P` parameter to see how it behaves with this change?